inputs/cluster/agn_triggering.in

```diff
@@ -0,0 +1,114 @@
+
+<comment>
+problem   = AGN Triggering Test
+
+<job>
+problem_id = cluster   # problem ID: basename of output filenames
+
+<parthenon/output1>
+file_type  = hst       # History data dump
+dt         = 1e-4      # time increment between outputs
+
+<parthenon/output2>
+file_type  = hdf5      # HDF5 data dump
+variables  = cons,prim      # Variables to be output
+dt         = 0.1     # Time increment between outputs
+id         = vars      # Name to append to output
+
+<parthenon/time>
+cfl_number = 0.3        # The Courant, Friedrichs, & Lewy (CFL) Number
+nlim       = -1         # cycle limit
+tlim       = 0.1       # time limit
+integrator  = vl2       # time integration algorithm
+perf_cycle_offset  = 10 # interval for stdout summary info
+
+
+<parthenon/mesh>
+refinement  = static
+nghost = 2
+
+nx1        = 64        # Number of zones in X1-direction
+x1min      =-0.1       # minimum value of X1
+x1max      = 0.1       # maximum value of X1
+ix1_bc     = outflow   # inner-X1 boundary flag
+ox1_bc     = outflow   # outer-X1 boundary flag
+
+nx2        = 64        # Number of zones in X2-direction
+x2min      =-0.1       # minimum value of X2
+x2max      = 0.1       # maximum value of X2
+ix2_bc     = outflow   # inner-X2 boundary flag
+ox2_bc     = outflow   # outer-X2 boundary flag
+
+nx3        = 64        # Number of zones in X3-direction
+x3min      =-0.1       # minimum value of X3
+x3max      = 0.1       # maximum value of X3
+ix3_bc     = outflow   # inner-X3 boundary flag
+ox3_bc     = outflow   # outer-X3 boundary flag
+
+<parthenon/static_refinement0>
+x1min = -0.025
+x1max =  0.025
+x2min = -0.025
+x2max =  0.025
+x3min = -0.025
+x3max =  0.025
+level = 2
+
+
+<parthenon/meshblock>
+nx1        = 8        # Number of zones in X1-direction
+nx2        = 8        # Number of zones in X2-direction
+nx3        = 8        # Number of zones in X3-direction
+
+<hydro>
+fluid = euler
+gamma = 1.6666666666666667 # gamma = C_p/C_v
+eos = adiabatic
+riemann = none
+reconstruction = dc
+calc_dt_hyp = true
+use_scratch = false
+scratch_level = 0 # 0 is actual scratch (tiny); 1 is HBM
+
+He_mass_fraction = 0.25
+
+<units>
+#Units parameters
+#Note: All other parameters for the cluster are in terms of these units
+code_length_cgs = 3.085677580962325e+24 # 1 Mpc in cm
+code_mass_cgs = 1.98841586e+47          # 1e14 Msun in g
+code_time_cgs = 3.15576e+16             # 1 Gyr in s
+
+<problem/cluster>
+
+<problem/cluster/gravity>
+#Define SMBH for Bondi accretion
+m_smbh = 1.0000000000000002e-06
```

Comment:
The mass is 1e-6 code-mass-units. 

Having mixed units in parameter files - some things in kpc, other in cgs, others in msun, etc. turned out to be a horrible mess in Enzo that lead to several user-errors on my part and previous developers. There were code-unit conversions everywhere and it made it difficult to debug/track parameters defined in the input file and where they were used in the code. With this implementation everything is in code-units.

I'm in favor of having everything in one system of units in parameter files. Thus, everything in the cluster input file is in code units except for the actual definition of code units.

I use a Jupyter notebook to generate and manage the input parameters for clusters, thus the floating point round-off. This has the added benefit to estimate data and memory usage when I create the file as well as potentially making plots of the initial cluster.

Reply:
Understood. As long as we're consistent and explicit I have not strong opinion on which way to go.
Especially for these more complex scenarios an external path (like the jupyter notebooks you prepared) may be more user friendly (also with regard to further analysis) than handling all the logic internally.
Having said that: How about to add the setup notebook also to the repo so that there's only one place where this lives (and needs to be maintained -- similar to the `generate forcing wavenumbers` python script I use for the turbulence sims).